router-tests/modules/stream_publish_test.go

```diff
@@ -0,0 +1,345 @@
+package module_test
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"go.uber.org/zap/zapcore"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/wundergraph/cosmo/router-tests/events"
+	stream_publish "github.com/wundergraph/cosmo/router-tests/modules/stream-publish"
+	"github.com/wundergraph/cosmo/router-tests/testenv"
+	"github.com/wundergraph/cosmo/router/core"
+	"github.com/wundergraph/cosmo/router/pkg/config"
+	"github.com/wundergraph/cosmo/router/pkg/pubsub/datasource"
+	"github.com/wundergraph/cosmo/router/pkg/pubsub/kafka"
+)
+
+func TestPublishHook(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Test Publish hook can't assert to mutable types", func(t *testing.T) {
```

Comment:
I'm putting this comment just on this test but my general feedback on the tests in router-tests is that it could be useful to add comments and the tests to give a bit more context on how each test adds variance in the test suite. The titles are rather short and I understand why, but as someone from outside, it's a bit hard to understand why each test is relevant and what particular (edge) case it's covering.